dotnet/src/dotnetcore/Providers/Messaging/GXAzureEventGrid/AzureEventGrid.cs

```diff
@@ -0,0 +1,338 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Messaging;
+using Azure.Messaging.EventGrid;
+using GeneXus.Messaging.Common;
+using GeneXus.Services;
+using GeneXus.Utils;
+
+namespace GeneXus.Messaging.GXAzureEventGrid
+{
+	public class AzureEventGrid : EventRouterBase, IEventRouter
+	{
+		public static string Name = "AZUREEVENTGRID";
+		private EventGridPublisherClient _client;
+		private string _endpoint;
+		private string _accessKey;
+		public AzureEventGrid() : this(null)
+		{
+		}
+		public AzureEventGrid(GXService providerService) : base(providerService)
+		{
+			Initialize(providerService);
+		}
+		private void Initialize(GXService providerService)
+		{
+			ServiceSettings serviceSettings = new(PropertyConstants.EVENT_ROUTER, Name, providerService);
+			_endpoint = serviceSettings.GetEncryptedPropertyValue(PropertyConstants.URI_ENDPOINT);
+			_accessKey = serviceSettings.GetEncryptedPropertyValue(PropertyConstants.ACCESS_KEY);
```

Comment:
Is this the recommended security approach to connect with Azure Services?

In AWS, Companies such as Globant, does not allow to connect to Any service using Security Credentials. They must use IAM Roles to access sensitive services. Is there anything equivalent in Azure?

Reply:
Maybe DefaultAzureCredential ?